moveit_core/robot_trajectory/test/test_robot_trajectory.cpp

```diff
@@ -75,7 +77,7 @@ class RobotTrajectoryTestFixture : public testing::Test
     double duration_from_previous = 0.1;
     std::size_t waypoint_count = 5;
     for (std::size_t ix = 0; ix < waypoint_count; ++ix)
-      trajectory->addSuffixWayPoint(robot_state_, duration_from_previous);
+      trajectory->addSuffixWayPoint(*robot_state_, duration_from_previous);
```

Comment:
Why did this API change, even though the function itself did not seem to?

Reply:
I stumbled over this question as well. Here is the answer:
Actually both API exists: taking a `RobotState&` and `RobotStatePtr`. While the former creates a new state copy, the latter uses the pointer as is. I guess, @v4hn wanted to ensure that a copy is created here.